From: Corey Bryant <corey.bryant@canonical.com>
Date: Tue, 6 Oct 2015 19:11:41 +0000 (-0400)
Subject: Iterate over copy of session.adapters keys in Python2/3
X-Git-Url: https://review.openstack.org/gitweb?p=openstack%2Fpython-keystoneclient.git;a=commitdiff_plain;h=c902b147cfbc44dfc140662cea6cc96f849b0b9b

Iterate over copy of session.adapters keys in Python2/3

Iterate over a copy of session.adapters keys in both Python 2.x and
Python 3.x.  In Python 3.x, keys() is not a copy, and therefore
items can't be popped from it while iterating.

Change-Id: Iaa2be0dc8ef26e51ce5e8f50049c9e8f84418ec0
Closes-Bug: #1483872
---

diff --git a/keystoneclient/session.py b/keystoneclient/session.py
index e3db3b9..ae640b6 100644
--- a/keystoneclient/session.py
+++ b/keystoneclient/session.py
@@ -135,7 +135,7 @@ class Session(object):
         if not session:
             session = requests.Session()
             # Use TCPKeepAliveAdapter to fix bug 1323862
-            for scheme in session.adapters:
+            for scheme in list(session.adapters):
                 session.mount(scheme, TCPKeepAliveAdapter())
 
         self.auth = auth
